src/operator/bufferToggle.ts

```diff
@@ -44,18 +46,18 @@ import {errorObject} from '../util/errorObject';
  * @owner Observable
  */
 export function bufferToggle<T, O>(openings: Observable<O>,
-                                   closingSelector: (value: O) => Observable<any>): Observable<T[]> {
+                                   closingSelector: (value: O) => SubscribableOrPromise<any> | void): Observable<T[]> {
```

Comment:
if selector throws in one execution branch like
```
if (...) {
throw  
}
```
 
then compiler will complain there isn't any return value in function. There's open issue to support this(https://github.com/Microsoft/TypeScript/issues/1042), until then without this typescript user can't throw or have to cast it into `any` explicitly.